### PR TITLE
fix(i18n): update Chinese translation of css grid

### PR DESCRIPTION
**Pull request by ZhichengChen**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/align-an-item-vertically-using-align-self.chinese.md [hunk 1]
```diff
@@ -2,23 +2,28 @@
...
-localeTitle: 使用align-self垂直对齐项目
+videoUrl: 'https://scrimba.com/p/pByETK/cmzd4fz'
+forumTopicId: 301123
+localTitle: 使用 align-self 垂直对齐项目
 ---
 
 ## Description
-<section id="description">正如您可以水平对齐项目一样，也可以垂直对齐项目。为此，您可以在项目上使用<code>align-self</code>属性。此属性接受来自上一次挑战的所有与<code>justify-self</code>相同的值。 </section>
+<section id='description'>
+正如能设置网格项沿行轴对齐方式一样，也可以设置网格项沿列轴对齐方式：你可以对网格项使用<code>align-self</code>属性。对于该属性，你能使用在上一个挑战中可用于<code>justify-self</code>属性的任一个值。
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/align-an-item-vertically-using-align-self.chinese.md]: "也可以设置网格项沿列轴对齐方式" -> "也可以设置网格项沿列轴对齐"

**Comment by RandellDawson** [comment]
@Becavalier How do these changes look now?

**Comment by Becavalier** [comment]
@RandellDawson Not fully ready yet, may need more revisions.

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md [hunk 2]
```diff
@@ -2,23 +2,48 @@
...
+</section>
 
 ## Tests
 <section id='tests'>
 
 ```yml
 tests:
-  - text: <code>container</code>类应具有<code>grid-template-columns</code>属性，该属性设置为重复3列，宽度为<code>1fr</code> 。
-    testString: 'assert(code.match(/.container\s*?{[\s\S]*grid-template-columns\s*?:\s*?repeat\s*?\(\s*?3\s*?,\s*?1fr\s*?\)\s*?;[\s\S]*}/gi), "<code>container</code> class should have a <code>grid-template-columns</code> property that is set to repeat 3 columns with the width of <code>1fr</code>.");'
+  - text: '<code>container</code>类应该有<code>grid-template-columns</code>属性且设置重复 3 列，宽为<code>1fr</code>。'
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md]: "应该有" -> "应该带有"
"设置" -> "设置为"

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md [hunk 3]
```diff
@@ -2,23 +2,48 @@
...
+    testString: 'assert(code.match(/.container\s*?{[\s\S]*grid-template-columns\s*?:\s*?repeat\s*?\(\s*?3\s*?,\s*?1fr\s*?\)\s*?;[\s\S]*}/gi), ''<code>container</code>类应该有<code>grid-template-columns</code>属性且设置重复 3 列，宽为<code>1fr</code>。'');'
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md]: "应该有" -> "应该带有"
"设置" -> "设置为"

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md [hunk 4]
```diff
@@ -2,23 +2,48 @@
...
+grid-template-columns: 1fr 50px 1fr 50px 20px;
+```
+
+<strong>注意：</strong><br><code>1fr 50px</code>重复了两次，后面跟着 20px。
+</section>
 
 ## Instructions
-<section id="instructions">使用<code>repeat</code>从<code>grid-template-columns</code>属性中删除重复。 </section>
+<section id='instructions'>
+用<code>repeat</code>代替<code>grid-template-columns</code>属性的值中的重复代码。
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md]: "属性的值" -> "属性值"

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md [hunk 5]
```diff
@@ -2,23 +2,36 @@
...
+}
+```
+
+这样，类名为<code>item1</code>的网格项就被放到了<code>header</code>区域里。这种情况下，网格项将使用整个顶行，因为这一行被名为 header 区域。
+</section>
 
 ## Instructions
-<section id="instructions">使用<code>grid-area</code>属性将具有<code>item5</code>类的元素放置在<code>footer</code>区域中。 </section>
+<section id='instructions'>
+请使用<code>grid-area</code>属性，把类为<code>item5</code>元素放到<code>footer</code>区域。
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md]: "元素" -> "的元素"

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md [hunk 6]
```diff
@@ -2,23 +2,36 @@
...
-localeTitle: 使用网格区域属性在网格区域中放置项目
+videoUrl: 'https://scrimba.com/p/pByETK/cRrqmtV'
+forumTopicId: 301132
+localTitle: 使用 grid-area 属性将项目放置在网格区域中
 ---
 
 ## Description
-<section id="description">为网格容器创建区域模板后，如上一个挑战所示，您可以通过引用您提供的名称将项目放入自定义区域。为此，您可以在项目上使用<code>grid-area</code>属性，如下所示： <blockquote> .item1 {grid-area：header; } </blockquote>这使网格知道您希望<code>item1</code>类进入名为<code>header</code>的区域。在这种情况下，该项将使用整个顶行，因为整行被命名为标题区域。 </section>
+<section id='description'>
+像上一个挑战那样，在为网格容添加区域模板后，你可以通过添加你定义的名称将网格项放入自定义区域。为此，你需要对网格项使用<code>grid-area</code>：
```
Comment (Becavalier) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md]: "在为网格容添加区域模板后" -> "在为网格添加区域模板后"

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md [hunk 7]
```diff
@@ -35,16 +48,16 @@ tests:
...
+  
   .item5 {
     background: PaleGreen;
     grid-column: 2 / 4;
-    /* add your code below this line */
-
-
-    /* add your code above this line */
+    /* 请在本行以下添加你的代码 */
+
```
[the remark this answers is not in the document]
Comment (ZhichengChen) [reply curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md]: Do you mean the extra rows under "/* 请在本行以下添加你的代码 */"? I think it reminds the user to add code in the blank row, and it's the same as the English version.

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md [hunk 8]
```diff
@@ -2,23 +2,40 @@
...
+</section>
 
 ## Tests
 <section id='tests'>
 
 ```yml
 tests:
-  - text: <code>item5</code>类应具有<code>grid-area</code>属性，使其位于第三和第四水平线之间以及第一和第四垂直线之间。
-    testString: 'assert(code.match(/.item5\s*?{[\s\S]*grid-area\s*?:\s*?3\s*?\/\s*?1\s*?\/\s*?4\s*?\/\s*?4\s*?;[\s\S]*}/gi), "<code>item5</code>类应具有<code>grid-area</code>属性，使其位于第三和第四水平线之间以及第一和第四垂直线之间。");'
+  - text: '<code>item5</code>类应该有<code>grid-area</code>属性且值为<code>3/1/4/4</code>。'
```
[the remark this answers is not in the document]
Comment (ZhichengChen) [reply curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md]: Yep, I think so.

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md [hunk 9]
```diff
@@ -2,23 +2,36 @@
...
-localeTitle: 使用网格区域属性在网格区域中放置项目
+videoUrl: 'https://scrimba.com/p/pByETK/cRrqmtV'
+forumTopicId: 301132
+localeTitle: 使用 grid-area 属性将项目放置在网格区域中
 ---
 
 ## Description
-<section id="description">为网格容器创建区域模板后，如上一个挑战所示，您可以通过引用您提供的名称将项目放入自定义区域。为此，您可以在项目上使用<code>grid-area</code>属性，如下所示： <blockquote> .item1 {grid-area：header; } </blockquote>这使网格知道您希望<code>item1</code>类进入名为<code>header</code>的区域。在这种情况下，该项将使用整个顶行，因为整行被命名为标题区域。 </section>
+<section id='description'>
+像上一个挑战那样，在为网格添加区域模板后，你可以通过添加你定义的名称将网格项放入自定义区域。为此，你需要对网格项使用<code>grid-area</code>：
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md]: 你可以通过添加你定义的名称将网格项放入自定义区域 => 你可以通过引用你所定义的区域的名称，将元素放入相应的区域

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md [hunk 10]
```diff
@@ -2,23 +2,36 @@
...
+
+```css
+.item1 {
+  grid-area: header;
+}
+```
+
+这样，类名为<code>item1</code>的网格项就被放到了<code>header</code>区域里。这种情况下，网格项将使用整个顶行，因为这一行被名为 header 区域。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/place-items-in-grid-areas-using-the-grid-area-property.chinese.md]: 在这个示例中，网格项将占用整个顶行，因为这一整行都被命名为 header 区域。

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md [hunk 11]
```diff
@@ -2,23 +2,48 @@
...
+videoUrl: 'https://scrimba.com/p/pByETK/cQvqyHR'
+forumTopicId: 301133
+localeTitle: 使用 repeat 函数减少重复
 ---
 
 ## Description
-<section id="description">使用<code>grid-template-columns</code>和<code>grid-template-rows</code>定义网格结构时，为所创建的每个行或列输入了一个值。让我们说你想要一个100行相同高度的网格。单独插入100个值是不太实际的。幸运的是，有一种更好的方法 - 使用<code>repeat</code>函数指定要重复列或行的次数，后跟逗号和要重复的值。这是一个创建100行网格的示例，每行高度为50px。 <blockquote> grid-template-rows：repeat（100,50px）; </blockquote>您还可以使用repeat函数重复多个值，并在定义网格结构时将该函数插入其他值中。这就是我的意思： <blockquote> grid-template-columns：repeat（2,1fr 50px）20px; </blockquote>这意味着： <blockquote> grid-template-columns：1fr 50px 1fr 50px 20px; </blockquote> <strong>注意</strong> <br> <code>1fr 50px</code>重复两次，然后是20px。 </section>
+<section id='description'>
+当使用<code>grid-template-columns</code>和<code>grid-template-rows</code>定义网格结构时，你需要为添加的每一行和每一列都输入一个值。
+如果要添加带 100 行高度相同的网格，单独放入 100 个值不太实际。幸运的是，有一种更好的方法——使用<code>repeat</code>方法指定行或列的重复次数，后面加上逗号以及需要重复的值。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/reduce-repetition-using-the-repeat-function.chinese.md]: 如果要添加带 100 行高度相同的网格，单独放入 100 个值不太实际 => 如果一个网格共有 100 行，每行高度相同，就得输入 100 个值，这不太实际

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md [hunk 12]
```diff
@@ -2,23 +2,36 @@
...
-localeTitle: 使用网格行控制间距
+videoUrl: 'https://scrimba.com/p/pByETK/c9WBLU4'
+forumTopicId: 301137
+localeTitle: 使用 grid-row 来控制剩余部分
 ---
 
 ## Description
-<section id="description">当然，像网格列一样，您也可以使一个项目同时占据多个网格行。您可以使用<code>grid-row</code>属性来定义项目开始和结束的网格行。 </section>
+<section id='description'>
+当然，你可以像列一样使网格项跨越多行。对于一个网格项，你可以用<code>grid-row</code>属性来确定开始和结束的水平线。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md]: 当然，和设置一个网格项占用多列一样，你也可以设置它占用多行。你可以使用<code>grid-row</code>属性来定义一个网格项开始和结束的水平线。

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.chinese.md [hunk 13]
```diff
@@ -2,23 +2,47 @@
 id: 5a90372638fddaf9a66b5d38
 title: Use grid-column to Control Spacing
 challengeType: 0
-videoUrl: ''
-localeTitle: 使用grid-column(网格列)控制间距
+videoUrl: 'https://scrimba.com/p/pByETK/cnzkDSr'
+forumTopicId: 301136
+localeTitle: 使用 grid-column 来控制剩余部分
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.chinese.md]: spacing 在这里的意思是“空间大小”？

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md [hunk 14]
```diff
@@ -2,23 +2,36 @@
 id: 5a90373638fddaf9a66b5d39
 title: Use grid-row to Control Spacing
 challengeType: 0
-videoUrl: ''
-localeTitle: 使用网格行控制间距
+videoUrl: 'https://scrimba.com/p/pByETK/c9WBLU4'
+forumTopicId: 301137
+localeTitle: 使用 grid-row 来控制剩余部分
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.chinese.md]: spacing 在这里的意思是“空间大小”？

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md [hunk 15]
```diff
@@ -2,23 +2,40 @@
...
 ## Description
-<section id="description">您在上一次挑战中学习的<code>grid-area</code>属性可以以其他方式使用。如果您的网格没有可引用的网格区域模板，您可以当场创建一个网格区域来放置项目，例如： <blockquote> item1 {grid-area：1/1/2/4; } </blockquote>这个区域用了您之前刚学过的网格行号来定义项目的地点。上例中的数字的意思是： <blockquote>grid-area (网格区域)：水平线开始于/垂直线开始于/水平线结束于/垂直线结束于; </blockquote>因此，示例中的项目将使用第一个网格横线和第2个网格横线之间的网格行以及第一个网格纵线和第四个网格纵线之间的网格列。 </section>
+<section id='description'>
+你在上一次挑战中学到的<code>grid-area</code>属性有另一种使用方式。如果网格中没有定义区域模板，你也可以像这样为它添加一个模板：
+
+```css
+item1 { grid-area: 1/1/2/4; }
+```
+
+这里使用了你之前学习的线号来定义网格项的区域。上例中数字代表这些值：
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md]: 线号 => 网格线编号

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md [hunk 16]
```diff
@@ -2,23 +2,40 @@
...
+
+```css
+grid-area: horizontal line to start at / vertical line to start at / horizontal line to end at / vertical line to end at;
+```
+
+因此，示例中的网格项将占用第 1 条和第 2 条水平线之间的行及第 1 条和第 4 条垂直线之间的列。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md]: 因此，示例中的网格项将占用第 1 条和第 2 条水平线之间的行及第 1 条和第 4 条垂直线之间的列 => 因此，示例中的网格项将占用第 1 条水平网格线（起始）和第 2 条水平网格线（终止）之间的行，及第 1 条垂直网格线（起始）和第 4 条垂直网格线（终止）之间的列。

File: curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md [hunk 17]
```diff
@@ -2,23 +2,40 @@
...
+</section>
 
 ## Instructions
-<section id="instructions">使用<code>grid-area</code>属性，将带<code>item5</code>class的元素放置到第三和第四条网格横线之间以及第一条和第四条网格纵线之间。 </section>
+<section id='instructions'>
+请用<code>grid-area</code>属性将类为<code>item5</code>的元素放置在第 3 条和第 4 条水平线及第 1 条和第 4 条垂直线之间的区域内。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/01-responsive-web-design/css-grid/use-grid-area-without-creating-an-areas-template.chinese.md]: 水平线 => 水平网格线
垂直线 => 垂直网格线

**Review by miyaliu666 — COMMENTED** [review]
我检查了翻译，没有检查代码。谢谢！

**Comment by RandellDawson** [comment]
@ZhichengChen In case you did not know, you can remove all the assert messages as the tester uses the `text` instead of the assert message.  This will clean up the `testString`s a lot.

**Comment by ZhichengChen** [comment]
> @ZhichengChen In case you did not know, you can remove all the assert messages as the tester uses the `text` instead of the assert message. This will clean up the `testString`s a lot.

OK, thank you.

**Comment by miyaliu666** [comment]
Not sure why 3 Project Euler problems are added to this pr @ZhichengChen

**Review by miyaliu666 — APPROVED** [review]

**Comment by scissorsneedfoodtoo** [comment]
Everything LGTM. Thank you for all of your hard work @ZhichengChen :1st_place_medal: : :100:

**Pull request merged**